### ISSUE TITLE
Commands plugins

### ISSUE THREAD

**Issue by fpliger** [Anaconda-Platform/anaconda-project#89]
This is a extended propsal for #32 and somehow something related to #86 but too much (not a requirement for each other)

The idea is to provide a simple plugin system that allows to plug custom command types (as the `notebook` or the `bokeh_app` commands) via modules (or packages) into a pre-defined directory (i.e. `<ENV_ROOT>\share\anaconda-project\plugins`). `anaconda-project` would check this path (or others if we want to support a list of pre-defined paths) and load the new plugin if that plugin-id is called in the project file.

[Havoc proposal] For example, for commands it could be something like a "get_command_handler" function that might return a thing like:
```
class NbApiCommandHandler:
   field_names = ("nbapi",)
   def parse(self, ...):
        """Find problems when loading anaconda-project.yml and returns a command object"""
   def transform_args(self, ...):
        ...
```

**Comment by fpliger** [comment]
One design decision is on how plugins can be discoverable so that `anaconda-project` can register new plugins (or to put if differently, can find registered plugins). The initial decision with @havocp was to just scan for modules/packages in a predefined set of paths (that could be managed seperately) and load plugins that implements the plugins API. 

This idea is fast and easy to implement and do not require plugins authors to care about packaging too much but it is also a bit hacky and less "standard", but sort of "ok-ish" as far as it is well documented. 

As much as I don't like `entry_points` it really seems that least bitter solution ... so it there are no big things that I'm not foreseeing, I'd change the plugins scan/registering design to go with `entry_points`...Any thoughts?

**Comment by havocp** [comment]
does entry points let someone bundle a plugin in their project?

I don't know a lot about entry points but have some impression the setup.py thing is a little more for a developer persona vs a data scientist persona

**Comment by fpliger** [comment]
Yes, correct. The idea of giving precedence to `entry_points` is that is is easier to package and distribute thus it is easier to already build a few plugins that match the major cmds that we can think about (i.e., flask, tornado, etc...). 

We probably want to have both designs honestly.. :)

**Comment by mcg1969** [comment]
Thanks for opening this issue, and apologies for the long delay in responding.

Anaconda Project has not been actively developed in some time and has largely been superseded by newer alternatives:

* [pixi](https://pixi.sh/) and [uv](https://docs.astral.sh/uv/) represent the first truly successful successors.
* [conda-workspaces](https://github.com/conda-incubator/conda-workspaces) is a promising pixi-compatible alternative for users who prefer to stay closer to the conda ecosystem.

Going forward, work on this repository is focused on critical bug fixes and on tooling to help users migrate existing Anaconda Project projects to pixi (and, by extension, conda-workspaces). See the [Pixi support](https://anaconda-project.readthedocs.io/en/latest/pixi-support.html) page in our documentation for details on the `anaconda-project export-pixi` command, which can convert an existing `anaconda-project.yml` into a `pixi.toml` for you.

The `anaconda-project` package itself will remain available, so existing projects relying on its stable feature set can continue to be supported. However, we do not anticipate further enhancement work, and for that reason we are closing out the bulk of our open issue backlog — including this one.

If this issue describes a critical regression in the stable feature set (as opposed to an enhancement or an edge case in a feature we no longer plan to expand) and you have a minimal reproducer against a current release, please feel free to reopen and we will take another look.

Thank you for your interest in Anaconda Project.
